cmd/swatch-time/swatch-time.go

```diff
@@ -23,58 +42,86 @@ func main() {
 			flagSet.PrintDefaults()
 			fmt.Fprint(output, "(no flags defaults to centibeat format @000.00)\n")
 		}
-		raw, standard, precise, date bool
 	)
+	flagSet.Usage = usage
 	const (
-		rawName      = "r"
-		standardName = "s"
+		rawName       = "r"
+		rawUsage      = "use raw float format @000.000000"
+		standardName  = "s"
+		standardUsage = "use Swatch standard format @000"
 	)
-	flagSet.Usage = usage
-	flagSet.BoolVar(&raw, rawName, false, "use raw float format @000.000000")
-	flagSet.BoolVar(&standard, standardName, false, "use Swatch standard format @000")
-	flagSet.BoolVar(&precise, "p", false, "use a more precise calculation method")
-	flagSet.BoolVar(&date, "d", false, "print date as well")
+	var (
+		rawFlag, standardFlag bool
+		handleLayout          = func(parameter, layout string) error {
+			if rawFlag && standardFlag {
+				return fmt.Errorf(
+					"cannot combine -%s and -%s flags",
+					rawName, standardName,
+				)
+			}
+			useLayout, err := strconv.ParseBool(parameter)
+			if err != nil {
+				return err
+			}
+			if useLayout {
+				set.layout = layout
+			}
+			return nil
+		}
```

Comment:
Yeah, to be honest I hastily just split the logic without treating the command with as much respect as the library itself.

I moved this into a method, where the scope should help with locality, and the method's name `registerLayoutFlags` should add some context.

Also changed the closure name to `parseFormatFlag`, since it's parsing either of the format flags' `parameter` value, while also being called from `flag.Parse`.
`layout` is the value being applied to `settings` rather than being parsed or validated itself.

Maybe this name makes sense? Maybe it should be `parseLayoutFlag` for consistency, or something else.

---

I also considered if there was a way to eliminate the closed over bools, but I think the alternative I came up with is harder to read. I'm not sure if there's a clearer way to express the same thing.
<details><summary>An alternative:</summary>

```go
func (set *settings) registerLayoutFlags(flagSet *flag.FlagSet) {
	const (
		rawName        = "r"
		rawUsage       = "use raw float format @000.000000"
		rawLayout      = swatch.MicroBeats
		standardName   = "s"
		standardUsage  = "use Swatch standard format @000"
		standardLayout = swatch.Beats
	)
	parseFormatFlag := func(parameter, layout string) error {
		var (
			previous = set.layout
			wasSet   = previous == rawLayout || previous == standardLayout
			conflict = wasSet && previous != layout
		)
		if conflict {
			return fmt.Errorf(
				"cannot combine -%s and -%s flags",
				rawName, standardName,
			)
		}
		useLayout, err := strconv.ParseBool(parameter)
		if err != nil {
			return err
		}
		if useLayout {
			set.layout = layout
		}
		return nil
	}
	flagSet.BoolFunc(rawName, rawUsage, func(parameter string) error {
		return parseFormatFlag(parameter, rawLayout)
	})
	flagSet.BoolFunc(standardName, standardUsage, func(parameter string) error {
		return parseFormatFlag(parameter, standardLayout)
	})
}
```

</details>